### PR TITLE
Add e2e test framework

### PR DESCRIPTION
**Pull request by zhengxiexie**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -11,28 +12,36 @@ replace (
 require (
 	github.com/agiledragon/gomonkey v2.0.2+incompatible
 	github.com/agiledragon/gomonkey/v2 v2.4.0
```
Comment (TaoZou1) [inline go.mod:14]: This file includes lots of function to operate(create/update) resources like service, and in nsx-operator test, will be replaced by apply yaml.
Those operation function may be deleted.

File: test/e2e/fixtures.go [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+		cmd := fmt.Sprintf("kubectl -n %s logs --all-containers %s", nsName, podName)
+		stdout := runKubectl(cmd)
+		if stdout == "" {
+			return nil
+		}
+		w.WriteString(stdout)
+		return nil
+	}
+	// TODO: export Antrea log
+	data.forAllMatchingPodsInNamespace("app=antrea-interworking", interworkingNamespace, writePodLogs)
```
[the remark this answers is not in the document]
Comment (zhengxiexie) [reply test/e2e/fixtures.go]: ok

File: test/e2e/fixtures.go [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+}
+
+func deletePodWrapper(tb testing.TB, data *TestData, namespace, name string) {
+	tb.Logf("Deleting Pod '%s/%s'", namespace, name)
+	if err := data.deletePod(namespace, name); err != nil {
+		tb.Logf("Error when deleting Pod: %v", err)
+	}
+}
+
+func createTestAgnhostPods(tb testing.TB, data *TestData, num int, ns string, nodeName string) (
```
[the remark this answers is not in the document]
Comment (zhengxiexie) [reply test/e2e/fixtures.go:56]: removed

File: test/e2e/fixtures.go [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+			return nil
+		}
+		w.WriteString(stdout)
+		return nil
+	}
+	// TODO: export Antrea log
+	data.forAllMatchingPodsInNamespace("app=antrea-interworking", interworkingNamespace, writePodLogs)
+
+	// dump the output of "kubectl describe" for Antrea pods to disk.
+	data.forAllMatchingPodsInNamespace("app=antrea-interworking", interworkingNamespace, func(nodeName, podName, nsName string) error {
```
[the remark this answers is not in the document]
Comment (zhengxiexie) [reply test/e2e/fixtures.go]: ok

File: go.mod (hunk 1 above)
Comment (zhengxiexie) [reply go.mod:14]: which file ?

Comment (TaoZou1) [reply go.mod:14]: the file with prefix k8s,  it seems you have deleted it

**Review by TaoZou1 — DISMISSED** [review]

**Pull request closed without merging**